src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -496,10 +498,19 @@ private[table] class IndexedTableImpl(
           schema,
           optimizationOptions(options),
           append = true) {
-          val tableChanges = BroadcastedTableChanges(None, indexStatus, Map.empty, Map.empty)
-          val fileActions =
-            dataWriter.optimize(tableID, schema, revision, indexStatus, indexFiles)
-          (tableChanges, fileActions)
+
+          import indexFiles.sparkSession.implicits._
+          val removeFiles =
+            indexFiles.map(IndexFiles.toRemoveFile(dataChange = false)).collect().toIndexedSeq
+
+          val data = snapshot.loadDataframeFromData(indexFiles)
+
+          val (dataExtended, tableChanges) =
+            DoublePassOTreeDataAnalyzer.analyzeOptimize(data, indexStatus)
+
+          val newFiles = dataWriter.write(tableID, schema, dataExtended, tableChanges)
+          dataExtended.unpersist()
```

Comment:
Where do you `persist` the dataFrame? I would suggest to keep unpersist and persist operation in the same method

Reply:
if we unpersist a dataframe before the action is called, it is useless, as the computation will run on a uncached dataset.

Reply:
mm true. But it can be tricky to have a `persist` somewhere in the code that you might not be aware. Perhaps dividing more the code would be a solution. But we can leave it for a second step. 

Also, another question: the `persist` is used only for performance improvement, or to avoid that some recomputation on the data affects the final result?

Reply:
In the optimization, we are calculating twice the cubed of each element:
1. A first time to compute the new block size with the updated weight, 
2. A second time to write the data. 

I saw that in the code, we are using profusely the broadcast, but we never unpersist them, which is not ideal as we are using memory that might be better used for other scopes. I would keep this code as it is and open an Issue when describing the problem to find a more general solution.

Reply:
Seems good. Just add a line on the code referencing the new issue and the problem we will solve.